lib/src/ui.dart

```diff
@@ -148,42 +162,79 @@ class NetworkLoggerScreen extends StatelessWidget {
           ),
         ],
       ),
-      body: StreamBuilder(
-        stream: eventList.stream,
-        builder: (context, snapshot) => ListView.builder(
-          itemCount: eventList.events.length,
-          itemBuilder: enumerateItems<NetworkEvent>(
-            eventList.events,
-            (context, item) => ListTile(
-              key: ValueKey(item.request),
-              title: Text(
-                item.request!.method,
-                style: TextStyle(fontWeight: FontWeight.bold),
-              ),
-              subtitle: Text(
-                item.request!.uri.toString(),
-                maxLines: 1,
-                overflow: TextOverflow.ellipsis,
-              ),
-              leading: Icon(
-                item.error == null
-                    ? (item.response == null
-                        ? Icons.hourglass_empty
-                        : Icons.done)
-                    : Icons.error,
-              ),
-              trailing: AutoUpdate(
-                duration: Duration(seconds: 1),
-                builder: (context) => Text(_timeDifference(item.timestamp!)),
+      body: Column(
+        children: [
+          TextField(
+            controller: searchController,
+            onChanged: (text) {
+              eventList.updated(NetworkEvent());
+            },
+            autocorrect: false,
+            textAlignVertical: TextAlignVertical.center,
+            decoration: InputDecoration(
+              filled: true,
+              fillColor: Colors.white,
+              prefixIcon: Icon(
+                Icons.search,
+                color: Colors.black26,
               ),
-              onTap: () => NetworkLoggerEventScreen.open(
-                context,
-                item,
-                eventList,
+              suffix: StreamBuilder(stream: eventList.stream, builder: (context, snapshot){
+                return Text(getEvents().length.toString() + ' results');
```

Comment:
Calling `getEvents()` which does somewhat heavy work twice would cause performance issues. I would suggest wrapping `StreamBuilder` over both SearchBar and ListView to reduce `getEvents()` call count.